components/stacks-network/src/chains_coordinator.rs

```diff
@@ -543,45 +557,49 @@ pub async fn publish_stacking_orders(
     let stacks_node_rpc_url = format!("http://{}", &services_map_hosts.stacks_node_host);
 
     let mut transactions = 0;
-    let pox_info: PoxInfo = reqwest::get(format!("{}/v2/pox", stacks_node_rpc_url))
-        .await
-        .expect("Unable to retrieve pox info")
-        .json()
-        .await
-        .expect("Unable to parse contract");
-
-    for pox_stacking_order in devnet_config.pox_stacking_orders.iter() {
-        if pox_stacking_order.start_at_cycle - 1 == pox_info.reward_cycle_id {
-            let mut account = None;
-            let accounts_iter = accounts.iter();
-            for e in accounts_iter {
-                if e.label == pox_stacking_order.wallet {
-                    account = Some(e.clone());
-                    break;
-                }
-            }
-            let account = match account {
-                Some(account) => account,
-                _ => continue,
-            };
 
-            transactions += 1;
+    for (i, pox_stacking_order) in devnet_config.pox_stacking_orders.iter().enumerate() {
+        if pox_stacking_order.start_at_cycle - 1 != block.metadata.pox_cycle_index {
+            continue;
+        }
+
+        let pox_info: PoxInfo = reqwest::get(format!("{}/v2/pox", stacks_node_rpc_url))
+            .await
+            .expect("Unable to retrieve pox info")
+            .json()
+            .await
+            .expect("Unable to parse contract");
```

Comment:
Is it possible to move this call outside of the loop so we aren't fetching many times?

Reply:
It was before. But I figured it would be called _less often_ if we do it in the loop, after this check
```
if pox_stacking_order.start_at_cycle - 1 != block.metadata.pox_cycle_index {
      continue;
  }
 ```
 
 Instead of outside of the loop. 
 Unless if there is a huge amount of a `stacking_orders` but that's unlikely (and not the case with the default config).
 
 But this is really a small optimization. What do you think?
 I could comment though it if we decide to keep it here

Reply:
Ahh, I see. This is probably overkill, so I'll leave it to your discretion, but you could cache the value on the first fetch:
```rust
let mut fetched_pox_info: Option<PoxInfo> = None;
    for (i, pox_stacking_order) in devnet_config.pox_stacking_orders.iter().enumerate() {
        if pox_stacking_order.start_at_cycle - 1 != block.metadata.pox_cycle_index {
            continue;
        }

        fetched_pox_info = match fetched_pox_info {
            None => Some(
                reqwest::get(format!("{}/v2/pox", stacks_node_rpc_url))
                    .await
                    .expect("Unable to retrieve pox info")
                    .json()
                    .await
                    .expect("Unable to parse contract"),
            ),
            Some(i) => Some(i),
        };

        let pox_info = fetched_pox_info.unwrap();
   ...
```

I just don't love making requests in a loop if avoidable, though this probably isn't a big deal. Your call! 🤷

Reply:
Because I expect the stacking to happen at different blocks, in practice, the /v2/pox-2 call will only happen once at most (again, unless if the stacking orders are modified)
On one side I definitely see the code smell with that fetch in the loop, but in practice, the loop will mostly do one iteration. So I don’t know if I want to add this bit of complexity